onnxruntime/test/common/path_test.cc

```diff
@@ -53,17 +53,21 @@ TEST(PathTest, Parse) {
 #endif
 }
 
-#ifndef _WIN32
 TEST(PathTest, ParseFailure) {
     auto check_parse_failure =
         [](const std::string& path_string) {
         Path p{};
         EXPECT_FALSE(Path::Parse(ToPathString(path_string), p).IsOK());
       };
 
-    check_parse_failure("//root_name_without_root_dir");
-}
+#ifdef _WIN32
+    check_parse_failure(R"(\\server_name_no_separator)");
+    check_parse_failure(R"(\\server_name_no_share_name\)");
+    check_parse_failure(R"(\\server_name\share_name_no_root_dir)");
+#else  // POSIX
+    check_parse_failure("//root_name_no_root_dir");
```

Comment:
Why "//root_name_no_root_dir" isn't valid?

Reply:
what does it mean? counterexamples are welcome :)

Reply:
On Linux, multipe '/'s just equals to a single '/'
For example, 'a//b' is 'a/b'

Reply:
right. but two initial '/'s are treated differently. see the last paragraph of 4.11 here: https://pubs.opengroup.org/onlinepubs/009695399/basedefs/xbd_chap04.html#tag_04_11
i'm not sure how we want to handle this case. it says "implementation-defined". but this parsing logic expects something of the form: "//root_name/".